gef.py

```diff
@@ -3678,35 +3746,35 @@ def is_x86() -> bool:
 
 
 @lru_cache()
-def is_arch(arch: int) -> bool:
-    elf = gef.binary or get_elf_headers()
-    return elf.e_machine == arch
+def is_arch(arch: Elf.Abi) -> bool:
+    return arch in gef.arch.aliases
 
 
-def set_arch(arch: Optional[str] = None, default: Optional[str] = None) -> Architecture:
+def reset_architecture(arch: Optional[str] = None, default: Optional[str] = None) -> None:
     """Sets the current architecture.
     If an arch is explicitly specified, use that one, otherwise try to parse it
     out of the current target. If that fails, and default is specified, select and
     set that arch.
-    Return the selected arch, or raise an OSError.
+    Does not return but raise an exception if the architecture cannot be set.
```

Comment:
```suggestion
    Raise an exception if the architecture cannot be set.
    Does not return a value.
```

Your wording makes it sound like this function runs forever ("does not return").